src/components/Draggable/FilterDraggable.tsx

```diff
@@ -0,0 +1,219 @@
+
+  const [open, setOpen] = useState(false)
+
+  const [all, setAll] = useState<0 | 1 | 2>(
+    filterState.size === 0 ? 0 : filterState.size === filterValues.length ? 2 : 1
+  )
+
+  const [buttonRef, setButtonRef] = useState<HTMLButtonElement>()
+
+  const theme = useTheme()
+
+  const [{ isDragging }, drag] = useDrag({
+    item: { type, name: name, origin },
+    end: (_item, monitor) => {
+      if (monitor.getDropResult() != null) onDragEnd()
+    },
+    collect: (monitor) => ({
+      isDragging: !!monitor.isDragging(),
+    }),
+  })
+
+  const handleClick = () => (open ? handleClose() : setOpen(true))
+
+  const handleClose = () => {
+    setOpen(false)
+    setSearchedFilterSet(filterValues)
+  }
+
+  const handleSelect = (element: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    filterState.has(element) ? filterState.delete(element) : filterState.add(element)
+    handleFilterUpdate(name, new Set<string>(filterState))
+    if (filterState.size === 0) {
+      setAll(0)
+    } else if (filterState.size === filterValues.length) {
+      setAll(2)
+    } else {
+      setAll(1)
+    }
+  }
+
+  const handleKeyDown = (filterKey: keyof T) => (event: any) => {
+    const key = event.nativeEvent.key
+
+    let direction
+
+    switch (key) {
+      case 'ArrowRight':
+        direction = 'right'
+        break
+
+      case 'ArrowLeft':
+        direction = 'left'
+        break
+
+      case 'ArrowUp':
+        direction = 'up'
+        break
+
+      case 'ArrowDown':
+        direction = 'down'
+        break
+    }
+
+    onKeyNav(direction, origin, filterKey)
+    onDragEnd()
+  }
+
+  const handleSearch = () => (event: any) => {
+    const searchResults = new Array<string>()
+    const searchText: string = (event.currentTarget.value as string).toLocaleLowerCase()
+    filterValues.forEach((element: string) => {
+      const stringElement = element + ''
+      const loweredElement = stringElement.toLocaleLowerCase()
+      const found = loweredElement.search(searchText) !== -1
+      found && searchResults.push(element)
+    })
+    setSearchedFilterSet(searchResults)
+  }
+
+  const handleSelectAll = () => (event: any) => {
+    if (all === 2) {
+      setAll(0)
+      handleFilterUpdate(name, new Set<string>(new Set<string>()))
+    } else {
+      setAll(2)
+      handleFilterUpdate(name, new Set<string>(filterValues))
+    }
+  }
+
+  const Row = ({ index, style }) => {
+    const showTodos = searchedFilterSet.length === filterValues.length
+    if (index === 0 && showTodos) {
+      return (
+        <DropdownItem key='todos' css={styles.dropdownItem}>
+          <Checkbox label='Todos os itens' onChange={handleSelectAll()} checked={all === 2} indeterminate={all === 1} />
+        </DropdownItem>
+      )
+    }
+
+    const value: string | undefined | null = searchedFilterSet[showTodos ? index - 1 : index]
+
+    if (value && value.length > 0) {
+      const bigValue = value.length > 45
```

Comment:
Where this magic number came from?

Reply:
I talked to Batista, who developed the component, and he said that these numbers (bigValue, FixedSlizeList height and itemSize) were chosen arbitrarily.
Do you think it is necessary to change these values or something?